generators/typescript/utils/core-utilities/fetcher/src/fetcher/requestWithRetries.ts

```diff
@@ -0,0 +1,20 @@
+const INITIAL_RETRY_DELAY = 1;
+const MAX_RETRY_DELAY = 60;
+const DEFAULT_MAX_RETRIES = 2;
+
+export async function requestWithRetries(
+    requestFn: () => Promise<Response>,
+    maxRetries: number = DEFAULT_MAX_RETRIES
+): Promise<Response> {
+    let response: Response;
```

Comment:
Nit: should we make an initial request here? Would also allow us to nix the non-null assertion

I think it also makes it true to the number of retries, I think right now it does one less since the first is part of the loop

But I could be doing the math wrong

Reply:
Good catch! I added an initial request to `requestWithRetries` and updated tests to reflect the additional call: https://github.com/fern-api/fern/pull/3977/commits/bf1ab2ae898e5ea2bf67cb87b8b5514b75296398